src/features/abi-methods/components/decoded-abi-method-arguments.tsx

```diff
@@ -5,15 +5,18 @@ import { TransactionLink } from '@/features/transactions/components/transaction-
 import { AccountLink } from '@/features/accounts/components/account-link'
 import { ApplicationLink } from '@/features/applications/components/application-link'
 import { AssetIdLink } from '@/features/assets/components/asset-link'
+import { getAbiMethodRepresentation } from '@/features/abi-methods/mappers'
 
 export function DecodedAbiMethodArguments({ method }: { method: AbiMethod }) {
+  const methodRepresentation = useMemo(() => getAbiMethodRepresentation(method), [method])
```

Comment:
This mapping logic in `getAbiMethodRepresentation` is quite expensive because we have to recursively build the arguments. Therefore, this component and `DecodedAbiMethodReturnValue` should be refactored to received `AbiMethodArgumentRepresentation` and `AbiMethodReturnRepresentation` respectively so that we don't need to run the mapping logic twice.